test/preview/components/routers/test_file_router.py

```diff
@@ -40,21 +41,50 @@ def test_run(self, preview_samples_path):
             preview_samples_path / "images" / "apple.jpg",
         ]
 
-        router = FileExtensionRouter(mime_types=["text/plain", "audio/x-wav", "image/jpeg"])
-        output = router.run(paths=file_paths)
+        router = FileTypeRouter(mime_types=["text/plain", "audio/x-wav", "image/jpeg"])
+        output = router.run(sources=file_paths)
         assert output
         assert len(output["text/plain"]) == 2
         assert len(output["audio/x-wav"]) == 1
         assert len(output["image/jpeg"]) == 1
         assert not output["unclassified"]
 
+    @pytest.mark.unit
+    def test_run_with_bytestreams(self, preview_samples_path):
+        """
+        Test if the component runs correctly with ByteStream inputs.
+        """
+        file_paths = [
+            preview_samples_path / "txt" / "doc_1.txt",
+            preview_samples_path / "txt" / "doc_2.txt",
+            preview_samples_path / "audio" / "the context for this answer is here.wav",
+            preview_samples_path / "images" / "apple.jpg",
+        ]
+        mime_types = ["text/plain", "text/plain", "audio/x-wav", "image/jpeg"]
+        # Convert file paths to ByteStream objects and set metadata
+        byte_streams = []
+        for path, mime_type in zip(file_paths, mime_types):
+            stream = ByteStream(path.read_bytes())
+
+            stream.metadata["content_type"] = mime_type
+
+            byte_streams.append(stream)
+
+        router = FileTypeRouter(mime_types=["text/plain", "audio/x-wav", "image/jpeg"])
+        output = router.run(sources=byte_streams)
+        assert output
+        assert len(output["text/plain"]) == 2
+        assert len(output["audio/x-wav"]) == 1
+        assert len(output["image/jpeg"]) == 1
+        assert not output.get("unclassified", None)
```

Comment:
Just for completeness, let's also add a ByteStream with no `content_type` and test that it goes under `unclassified`